distributed/comm/utils.py

```diff
@@ -1,43 +1,88 @@
 from __future__ import print_function, division, absolute_import
 
+from concurrent.futures import ThreadPoolExecutor
 import logging
 import socket
 
+from tornado import gen
+
 from .. import protocol
-from ..utils import get_ip, get_ipv6
+from ..compatibility import finalize
+from ..sizeof import sizeof
+from ..utils import get_ip, get_ipv6, nbytes
 
 
 logger = logging.getLogger(__name__)
 
 
+# Offload (de)serializing large frames to improve event loop responsiveness
+FRAME_OFFLOAD_THRESHOLD = 10 * 1024 ** 2   # 10 MB
+
+
+class Offloader(object):
+
+    def __init__(self):
+        self.executor = ThreadPoolExecutor(max_workers=1)
```

Comment:
We might want a couple of workers here?  We do accept multiple messages concurrently.  Having multiple threads here might help fast messages get through in the presence of large messages.